Adds instruction for verifying Stylus contracts on Arbiscan (#1684)
OffchainLabs/arbitrum-docs: arbitrum-docs/stylus/how-tos/verifying-contracts-arbiscan.mdx

---
title: 'How to verify Stylus contracts on Arbiscan'
description: 'This page provides a step-by-step guide to verifying Stylus contracts on Arbiscan, including contract details, source code submission, and handling previously verified contracts.'
author: mahsamoosavi
sme: mahsamoosavi
target_audience: 'Developers deploying smart contracts using Stylus'
content_type: how-to
sidebar_position: 1
---

import PublicPreviewBannerPartial from '../../partials/_public-preview-banner-partial.md';

<PublicPreviewBannerPartial />

> For example, let's consider the [English Auction Stylus contract](https://github.com/OffchainLabs/stylus-english-auction), which has been verified on Arbitrum Sepolia. You can view the verified contract [here](https://sepolia.arbiscan.io/address/0xe85a046fd3ea22ceeb3caef3a0d38123eecbe3ca).
> 
> You can also see a list of all Stylus contracts verified on Arbiscan by visiting:
> 
> - [Verified Stylus Contracts on Arbitrum One](https://arbiscan.io/contractsVerified?filter=stylus).
> - [Verified Stylus Contracts on Arbitrum Sepolia](https://sepolia.arbiscan.io/contractsVerified?filter=stylus).

Reviewer: ```suggestion
This how-to will show you how to verify deployed contracts using Arbiscan, Arbitrum's block explorer.

Here's an example of a verified contract: the [English Auction Stylus contract](https://github.com/OffchainLabs/stylus-english-auction), which has been verified on Arbitrum Sepolia. You can view the verified contract [here](https://sepolia.arbiscan.io/address/0xe85a046fd3ea22ceeb3caef3a0d38123eecbe3ca).

You can also see a list of all Stylus contracts verified on Arbiscan by visiting:

- [Verified Stylus Contracts on Arbitrum One](https://arbiscan.io/contractsVerified?filter=stylus).
- [Verified Stylus Contracts on Arbitrum Sepolia](https://sepolia.arbiscan.io/contractsVerified?filter=stylus).
```

> 
> ## Step 1: Navigate to the verification page
> 
> You have two options to access the contract verification page on Arbiscan:
> 
> 1. **Direct link:** Visit [Arbiscan Verify Contract](https://arbiscan.io/verifyContract) to go directly to the verification form. This option is ideal if you already have the contract address and details ready.
> 2. **From the contract page:** If you're viewing the contract's page on Arbiscan:
>    - Go to the **Contract** tab.
>    - Click on **Verify and Publish**.
> 
> ![Verify through the contract page](../assets/arbiscan-verification-1.png)
> 
> Both methods will take you to the contract verification form, where you can proceed to the next step.
> 

Reviewer: ```suggestion

Here are the steps to take to verify a contract on Arbiscan:

## Step 1: Navigate to the verification page

You have two options to access the contract verification page on Arbiscan:

1. **Direct link:** Visit [Arbiscan Verify Contract](https://arbiscan.io/verifyContract) to go directly to the verification form. This option is ideal if you already have the contract address and details ready.
2. **From the contract page:** If you're viewing the contract's page on Arbiscan:
   - Go to the **Contract** tab.
   - Click on **Verify and Publish**.

![Verify through the contract page](../assets/arbiscan-verification-1.png)

Both methods will take you to the contract verification form, where you can proceed to the next step.

```

> You have two options to access the contract verification page on Arbiscan:
> 
> 1. **Direct link:** Visit [Arbiscan Verify Contract](https://arbiscan.io/verifyContract) to go directly to the verification form. This option is ideal if you already have the contract address and details ready.
> 2. **From the contract page:** If you're viewing the contract's page on Arbiscan:
>    - Go to the **Contract** tab.
>    - Click on **Verify and Publish**.
> 
> ![Verify through the contract page](../assets/arbiscan-verification-1.png)
> 
> Both methods will take you to the contract verification form where you can proceed to the next step.
> 

Reviewer: ```suggestion
You have two options to access the contract verification page on Arbiscan:

1. **Direct link:** Visit [Arbiscan Verify Contract](https://arbiscan.io/verifyContract) to go directly to the verification form. This option is ideal if you already have the contract address and details ready.
2. **From the contract page:** If you're viewing the contract's page on Arbiscan:
   - Go to the **Contract** tab.
   - Click on **Verify and Publish**.

![Verify through the contract page](../assets/arbiscan-verification-1.png)

Both methods will take you to the contract verification form, where you can proceed to the next step.

```

> ## Step 2: Enter contract details
> 
> You will need to fill in the following fields on the contract verification page:
> 
> - **Contract address**: Enter the contract address you want to verify.
> - **Compiler type**: Select **Stylus** for Stylus contracts.
> - **Compiler version**: Choose the `cargo stylus` version that was used to deploy the contract.
> - **Open source license type**: Select the appropriate license for your contract.
> 
> ![Enter contract details](../assets/arbiscan-verification-2.png)
> 

Reviewer: ```suggestion
## Step 2: Enter the contract's details

You will need to fill in the following fields on the contract verification page:

- **Contract address**: Enter the contract address you want to verify.
- **Compiler type**: Select **Stylus** for Stylus contracts.
- **Compiler version**: Choose the `cargo stylus` version that was used to deploy the contract.
- **Open source license type**: Select the appropriate license for your contract.

![Enter contract details](../assets/arbiscan-verification-2.png)

```

> ## Step 3: Submit source code
> 
> After entering the contract details, you’ll need to provide the contract's source code:
> 
> - **Manual submission**: Copy and paste the source code into the provided text box.
> - **Fetch from GitHub (Recommended)**: It's recommended to use the **Fetch from Git** option, as it's easier and helps automate the process. However, note that contracts located in subdirectories of the repository cannot be verified. Ensure that the contract's code is placed directly in the root of the repository for verification to succeed.
> 
> ![Fetch source code](../assets/arbiscan-verification-3.png)

Reviewer: ```suggestion
## Step 3: Submit source code

After entering the contract details, you’ll need to provide the contract's source code:

- **Manual submission**: Copy and paste the source code into the provided text box.
- **Fetch from GitHub (Recommended)**: It's recommended to use the **Fetch from Git** option, as it's easier and helps automate the process. However, note that contracts located in subdirectories of the repository cannot be verified. Ensure that the contract's code is placed directly in the repository's root for verification to succeed.

![Fetch source code](../assets/arbiscan-verification-3.png)
```
